Osc hoverinfo

avih opened the pull request:
Three commits:
- Add `hoverinfo` framework - "status" at the osc title while hovering some osc elements.
- Add static info text for most buttons.
- Add dynamic seekbar info with target chapter.

The framework allows either static or dynamic text per element (or none). Info for disabled elements does show up - to allow discoverability, but it's grayed-out like the disabled element itself.

The second commit (static buttons text) is open to bikeshedding about format/texts and whether or not we want it also for trivial buttons (e.g. the play/pause button currently does have hoverinfo), but I suggest not to dwell on it too much because it's trivial to change at any time.

Personally I don't have a strong opinion on what the info texts should be exactly, other than it should be useful.

The third commit adds one dynamic info (the seekbar), but we can also consider dynamic info for simple buttons, for instance we could do this for the play/pause button:
```lua
ne.hoverinfo = function()
    return mp.get_property_native("pause") and "play" or "pause"
end
```

and similar for mute/unmute, enter/exit fullscreen, etc, though again, it's trivial to change the actual infos at any time.

If anyone wants to test it locally, this is `osc.lua` in mpv master with the 3 commits applied: https://0x0.st/-9Ae.lua  (`curl https://0x0.st/-9Ae.lua > ./osc.lua && mpv --no-osc --script=./osc.lua myclip.mkv`)

avih commented:
Unless there's objection or better suggestions, I think I'll change the static descriptions from these:
```
LMB: show media-title | RMB: show filename
(playlist) LMB: previous | RMB: show
(playlist) LMB: next | RMB: show
(play/pause) LMB: toggle
(seek back) LMB: 5 s | RMB: 30 s | shift+LMB: 1 frame
(seek forward) LMB: 10 s | RMB: 60 s | shift+LMB: 1 frame
(chapters) LMB: previous | RMB: show
(chapters) LMB: next | RMB: show
(audio tracks) LMB: next | RMB: previous | shift-LMB: show
(subtitle tracks) LMB: next | RMB: previous | shift-LMB: show
toggle fullscreen
(current position) LMB: toggle fractions display
(end-of-clip) LMB: toggle duration/remaining
(volume) LMB: toggle mute | wheel up/down: change volume
```

to these, which I think are more readable without parenthesis, and more consistent in the form of "subject/desc - left-click-action | RMB: ...":

```
Title - show media-title | RMB: show filename
Playlist - previous | RMB: list
Playlist - next | RMB: list
Play/Pause - toggle
Skip backward - 5 s | RMB: 30 s | shift+LMB: frame-back-step
Skip forward - 10 s | RMB: 60 s | shift+LMB: frame-step
Chapters - previous | RMB: list
Chapters - next | RMB: list
Audio tracks - next | RMB: previous | shift-LMB: list
Subtitle tracks - next | RMB: previous | shift-LMB: list
Fullscreen - toggle
Current position - toggle fractions display
End of clip - toggle duration/remaining
Volume - toggle mute | wheel up/down: change volume
```

avih commented:
Updated the following:
- First commit: typo (alphe -> alpha), few comments updates, don't display "cancel" for slider (mouse-down and then exit a button does cancel the button press, but for a slider apparently it doesn't cancel it).
- Second commit: updated static texts as described above.
- Third commit is unmodified.

Dudemanguy reviewed (approved):
Pretty cool idea. I didn't even know some of these right mouse button commands existed.

avih commented:
Updated: replaced the 3 commits with just one - which adds only the seekbar-chapters info.

For future reference, the original PR commits and the resulting `osc.lua` are in this zip: [mpv-hoverinfo-patchset-x4.zip](https://github.com/mpv-player/mpv/files/6682663/mpv-hoverinfo-patchset-x4.zip) - it also includes an additional 4th commit to select which elements would have the hover-info.

Reasoning:

After some discussion on irc, we concluded that the original patches in this PR did few things:
1. Useful dynamic info (the seekbar chapters) - can be used to make a decision where to seek.
2. Static help, which might be useful initially, but can become an annoyance quickly - and it's likely that many would want to turn it off at some point.
3. Allows potential dynamic info which is not really helpful to make a decision (maybe next chapter name over the next-chapter button, etc).

The first of those (seekbar chapters) is strictly useful and probably doesn't need an option.

The other two (static help and "cute" info) are not really helpful in the long run, and we might want to think of integrating them as part of some help system without ending up annoying by default. For instance, @ChrisK2 suggested a `?` button which could be used for GUI help. However, no one volunteeded for this so far.

So this PR now only does the seekbar chapters info, unconditionally.

avih commented:
Updated: revised `osc.rst` docs with the new title functionality.

Dudemanguy commented:
I was actually fine how it was initially, but this is still good anyway. Number 1 on that list is definitely useful.
